packages/perf/src/index.ts

```diff
@@ -0,0 +1,208 @@
+/**
+ * @packageDocumentation
+ *
+ * The `performanceService` implements the [perf protocol](https://github.com/libp2p/specs/blob/master/perf/perf.md), which is used to measure performance within and across libp2p implementations
+ * addresses.
+ *
+ * @example
+ *
+ * ```typescript
+ * import { createLibp2p } from 'libp2p'
+ * import { circuitRelayTransport } from 'libp2p/circuit-relay'
+ *
+ * const node = await createLibp2p({
+ *   service: [
+ *     perfService()
+ *   ]
+ * })
+ * ```
+ *
+ * The `measurePerformance` function can be used to measure the latency and throughput of a connection.
+ * server.  This will not work in browsers.
+ *
+ * @example
+ *
+ * ```typescript
+ * import { createLibp2p } from 'libp2p'
+ * import { circuitRelayServer } from 'libp2p/circuit-relay'
+ *
+ * const node = await createLibp2p({
+ *   services: [
+ *     perf: perfService()
+ *   ]
+ * })
+ *
+ * const connection = await node.dial(multiaddr(multiaddrAddress))
+ *
+ * await node.services.perf.measurePerformance(connection, BigInt(uploadBytes), BigInt(downloadBytes))
+ *
+ * ```
+ */
+
+import { logger } from '@libp2p/logger'
+import { anySignal } from 'any-signal'
+import { MAX_INBOUND_STREAMS, PROTOCOL_NAME, WRITE_BLOCK_SIZE } from './constants.js'
+import type { Connection } from '@libp2p/interface/src/connection/index.js'
+import type { Startable } from '@libp2p/interface/startable'
+import type { ConnectionManager } from '@libp2p/interface-internal/connection-manager'
+import type { IncomingStreamData, Registrar } from '@libp2p/interface-internal/registrar'
+import type { AbortOptions } from '@libp2p/interfaces'
+
+const log = logger('libp2p:perf')
+
+export const defaultInit: PerfServiceInit = {
+  protocolName: '/perf/1.0.0',
+  maxInboundStreams: 1 << 10,
+  maxOutboundStreams: 1 << 10,
```

Comment:
Are these even used? Nothing in the spec mentions "maxInboundStreams". I think we should get rid of them.